src/browser/components/molecules/Task/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react'
+import Layout from '../../molecules/Layout/index'
+import styles from './index.css'
+const moment = require('moment')
+
+
+
+export default class Task extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      complete: false
+    }
+    this.isCompleteHandler = this.isCompleteHandler.bind(this)
+  }
+
+  isCompleteHandler( event ){
+    const completeTaskDetails = {
+      method: 'POST',
+      mode: 'cors',
+      headers: new Headers({
+        'Accept': 'application/json, application/xml, text/plain, text/html, *.*',
+        'Content-Type': 'application/json',
+         }),
+      credentials: 'same-origin',
+      body: JSON.stringify({is_complete: true})
+    }
+    fetch( `http://noob.learnersguild.dev/api/task/${this.props.id}`, completeTaskDetails )
+    .then( resultPromise => {
+      return resultPromise.json()
+    })
+    .then( result => {
+      this.props.loadTasks()
+    })
+    .catch(err => {
+      console.log('Error marking task complete', err);
+      return err
+    })
+  }
+
+  taskStyles(){
```

Comment:
Since we are not using the `style` for `dueDateDiv` this function can be simplified. Another thing to keep in mind is that if you are writing helper functions, you should just pass the `props` or `state` as arguments to the function. That way the function can be made stateless(pure) and can be easily tested.

Something like this -
```javascript
function taskStyle(task) {
  if(task.is_complete) {
    return styles.complete
  } else {
    const diffInDays = moment(task.due_date).diff(moment(task.due_date))
    if(diffInDays < 0) {
      return styles.pastDue
    } else if (diffInDays > 4) {
      return styles.soonDue
    } else {
      return styles.nearDue
    }
  }
}
```